File: 프로그래머스/3/12938. 최고의 집합/최고의 집합.java
class Solution {
    public int[] solution(int n, int s) {
        int a = s / n;
        int b = s % n;
        if (a == 0){
            return new int[]{-1};
        }
        int[] answer = new int[n];
        for(int i = answer.length-1; i >= 0;i--){            
            answer[i] = a;
            if(b > 0){
                answer[i]++;
                b--;
            }
        }

        return answer;
    }
}

File: 프로그래머스/lv3/12938. 최고의 집합/최고의 집합.js
function solution(n, s) {
    if (s < n) return [-1];
    var answer = [];
    if (s % n == 0) {
        for (let i = 0; i < n; i++) {
            answer.push(s / n);
        }
    }
    else {
        for (let i = 0; i < n; i++) {
            answer.push(Math.floor(s / n + i));
        }
    }
    return answer;
}